components/core/CMakeLists.txt

```diff
@@ -136,6 +128,33 @@ else()
     endif()
 endif()
 
+# Determine the compatible fmt w.r.t to spdlog
+if(spdlog_VERSION VERSION_GREATER_EQUAL "1.15.0")
+    set(FMT_MIN_VERSION "11.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.12.0")
+    set(FMT_MIN_VERSION "10.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.11.0")
+    set(FMT_MIN_VERSION "9.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.9.0")
+    set(FMT_MIN_VERSION "8.0.0")
+else()
+    message(FATAL_ERROR "Unsupported spdlog version (${spdlog_VERSION}). Minimum supported version is 1.9.0.")
+endif()
+
+# Find fmt with the compatible version
+# NOTE:
+# - We only try to link to the static library
+find_package(fmt ${FMT_MIN_VERSION} REQUIRED)
```

Comment:
It's not enough to specify the minimum version since fmt's CMake scripts don't enforce compatibility with the specified version. E.g., if we set the minimum version to 8.0.0 but fmt v11 is installed, fmt's CMake scripts will find v11 and won't report an error. So in essence, the problem you experienced isn't solved.

Instead, we should *first* find both spdlog and fmt, *and then* determine if they're compatible.

Reply:
Added maximum version check in the latest revision.